plugins/dialog/plugin.js

```diff
@@ -1488,6 +1500,63 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 			}
 
 			this.fire( 'state', state );
+		},
+
+		/**
+		 * @inheritdoc CKEDITOR.dialog.modeledDialog#getModel
+		 */
+		getModel: function( editor ) {
+			var ret = this._.model;
+
+			// Always return cached model instance.
+			if ( this._.model ) {
+				return this._.model;
+			}
+
+			if ( this.definition.getModel ) {
+				ret = this.definition.getModel( editor ) || null;
+			}
+
+			if ( !ret ) {
+				ret = this.fire( 'getModel', {
+					model: ret
+				}, editor ).model;
+			}
+
+			return ret || null;
+		},
+
+		/**
+		 * @since 4.11.0
+		 * @param {CKEDITOR.dom.element/CKEDIOTR.plugins.widget/*} newModel Model to be set.
+		 */
+		setModel: function( newModel ) {
+			this._.model = newModel;
+		},
+
+		/**
+		 * Method tells whether the dialog is editing an existing element or adding a new one.
+		 *
+		 * In case if dialog does not
+		 *
+		 * @since 4.11.0
+		 * @param {CKEDITOR.editor} editor
+		 * @returns {Boolean} Returns `true` if dialog is editing content that already exists in the editor.
+		 */
+		isEditing: function( editor ) {
```

Comment:
Honestly, if it only returns `edit` and `create` events, enums seem to me like overkill.

Reply:
Sure, could you provide an example where `getMode` could return a different value than `edit` or `create`, so we won't implement the unused feature? If no, do you see more readable replacing all 
```javascript
 if ( dialog.isEditing( editor ) ) {
// do stuff
}
```
code samples with
```javascript
if ( dialog.getMode( editor ) === CKEDITOR.dialog.EDITING_MODE ) {
// do stuff
}
```
It would be nice to get a bit more info why you would like to use enum value than just beeing in favor :wink:

Reply:
I find enum-based solution more elegant, that's it. There is no strong logical argument for such solution and if we also take simplicity into the consideration, it clearly loses with `isEditing`.

In case of third state – `CKEDITOR.dialog.PENDING_MODE` ;) It could be used by dialogs that are asynchronously rendering its content. However not sure how helpful it would really be.

Reply:
> In case of third state – CKEDITOR.dialog.PENDING_MODE ;) It could be used by dialogs that are asynchronously rendering its content. However not sure how helpful it would really be.

I would say it's a strong argument because some plugins/widgets may be loaded asynchronously. In that case, they may require some fancy pending state. I know it's a bit exaggeration for today because we don't have any strong use case for now (maybe except mathjax plugin ? ), but the change is too simple to be sorry later.